<launch>

  <arg name="swap_lr"                   default="false"/>
  <arg name="disableRgb"                default="false"/>
  <arg name="disableLR"                 default="false"/>
  <arg name="field_of_view"             default="71.86"/>
  <arg name="baseline"                  default="7.5"/>
  <arg name="square_size_cm"            default="4"/>
  <arg name="marker_size_cm"            default="3.1"/>
  <arg name="squares_x"                 default="22"/>
  <arg name="squares_y"                 default="16"/>
  <arg name="usbMode"                   default="true"/>
  <arg name="depthai_path"              default="/Desktop/depthai"/>
  <arg name="calib_path"                default="/Desktop/calib_files"/>
  <arg name="log_path"                  default="/Desktop/logs"/>
  <arg name="brd"                       default="oak-d-lite"/>
  <arg name="enable_IMU_test"           default="false"/>
  <arg name="capture_service_name"      default="capture_checkerboard"/>
  <arg name="calibration_service_name"  default="calibrate_stereo"/>
  
  <node 
    name="depthai_calibration"
    pkg="calibration"
    type="depthai_calibration.py"
    respawn="false"
    output="screen"
    required="true">
    
    <param name="package_path"             value="$(find calibration)"/>
    <param name="swap_lr"                  value="$(arg swap_lr)"/>
    <param name="disableRgb"               value="$(arg disableRgb)"/>
    <param name="disableLR"                value="$(arg disableLR)"/>
    <param name="field_of_view"            value="$(arg field_of_view)"/>
    <param name="baseline"                 value="$(arg baseline)"/>
    <param name="square_size_cm"           value="$(arg square_size_cm)"/>
    <param name="marker_size_cm"           value="$(arg marker_size_cm)"/>
    <param name="squares_x"                value="$(arg squares_x)"/>
    <param name="squares_y"                value="$(arg squares_y)"/>
    <param name="usbMode"                  value="$(arg usbMode)"/>
    <param name="depthai_path"             value="$(arg depthai_path)"/>
    <param name="calib_path"               value="$(arg calib_path)"/>
    <param name="log_path"                 value="$(arg log_path)"/>
    <param name="brd"                      value="$(arg brd)"/>
    <param name="enable_IMU_test"          value="$(arg enable_IMU_test)"/>
    <param name="capture_service_name"     value="$(arg capture_service_name)"/>
    <param name="calibration_service_name" value="$(arg calibration_service_name)"/>
  </node>

  <!-- <node name="color_image_view" pkg="rqt_image_view" type="rqt_image_view" args="/color"/> -->


</launch> 